packages/flat-components/src/components/ClassroomPage/CloudRecordBtn/index.tsx

```diff
@@ -1,23 +1,35 @@
+import React, { useCallback, useState } from "react";
 import { observer } from "mobx-react-lite";
-import React from "react";
 import { useTranslation } from "react-i18next";
+
 import { TopBarRightBtn } from "../TopBar";
 import { SVGRecord, SVGRecordStop } from "../../FlatIcons";
+import { useSafePromise } from "../../../utils/hooks";
 
 export type CloudRecordBtnProps = {
     isRecording: boolean;
-    onClick: (evt: any) => void;
+    onClick: () => Promise<void>;
```

Comment:
`onClick` semantically should be an instant action. Mixing action with loading states seems to be an unconventional design.

Reply:
Otherwise we have to maintain the loading state in classRoomStore,

```jsx
<CloudRecordBtn
	loading={classRoomStore.isCloudRecordButtonLoading}
	onClick={classRoomStore.toggleRecord}
```

is that sound?